### ISSUE TITLE
NVDA Symbol Pronunciation Dialog (English) not always announcing symbol actually under edit

### ISSUE THREAD

**Issue by britechguy** [nvaccess/nvda#14040]
Problem Description
### Steps to reproduce: 
1. Hit NVDA + N, P, P [NVDA Menu, Preferences, Punctuation/Symbol Pronunciation]
2. In the filter box, enter "tick" (sans quotes)
3. Hit TAB once to land in the first returned result in the symbols list.
4. NVDA reads, "Symbols list in-word semi replacement colon tick semi level colon all preserve colon only below symbol's level 1 of 9"
Note: If I down arrow to the standalone single quote, rather than the in-word single quote, NVDA reads:  "semi Replacement colon tick semi Level colon all semi Preserve colon never  2 of 9"
### Actual Behavior: 
Documented in Problem description
### Expected Behavior: 
In that dialog, I would always expect the name of the character being "worked on" to be announced, but it is not being for the single quote nor for the bang when "bang" is the filter by.  In both cases the first thing announced is just "semi" with no indication of the character being worked on.  In addition, I would not expect either the separators between fields (e.g., semi) nor the terminator character for labels (colon) to be announced.  In the case of tick (and I'll use the naked one for this example) I would expect to hear, "tick, replacement tick, level all, preserve never 2 of 5"  I care about the character/symbol/punctuation I'm working on, the replacement string to be spoken, level, and preservation and always have to have the character being worked on announced.  I do not want or need punctuation that's a part of the list item labels (the colons) announced after the label itself nor do I need the list item separator, semi, announced since the announcement of the label itself makes it entirely clear where I am.  The critical problem is that the symbol/punctuation under edit is sometimes not announced at all.  It is consistently the case for tick and bang, it may occur for some others, but I did not attempt to test for every symbol or punctuation mark.  Many are working just fine.

#### NVDA Installed/Portable/Running from some other source:
Installed
#### NVDA Version: 
2022.2
#### Windows Version: 
Windows 10, Version 21H2, Build 19044.1889
#### Name and Version number of the other software in use when reproducing the issue: 
N/A
#### Other information about your system (e.g., Make & Model, Graphics Card, etc.): 
HP 15 series laptop with AMD A12-9700P APU

#### Does the issue still occur after restarting your computer?: 
Yes
#### Have you tried using any other versions of NVDA to reproduce the problem?  If so, please give the version number(s) and behavior(s): 
No
#### If you disable add-ons, is your problem still occurring?: 
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's Tools menu?: 
Yes, and machine was restarted afterward before checking the behavior again.

**Comment by lukaszgo1** [comment]
Have you tried this with different synthesizers - at least eSpeak and OneCore?

**Comment by britechguy** [comment]
I have only used Microsoft David (OneCore).  If this were an issue with the synth, I'd think that when reading character-by-character I'd get nothing for the tick when encountered, but that's not the case.  Regardless of whether the replacement is tick or single quote, this is announced when using NVDA to read character-by-character.  The same is true for bang.  It's only not reading in the Symbol Pronunciation Dialog.

**Comment by lukaszgo1** [comment]
I've tested this with the English eSpeak, and I cannot reproduce this. Can you confirm this affects only OneCore?

**Comment by britechguy** [comment]
I get the same result with eSpeak (using one of the American English male voices).  Nothing but "semi" when it's reading the character being worked on when that character is the tick.  I did not test for bang once the test for tick gave me the same result with eSpeak.

**Comment by lukaszgo1** [comment]
What is your punctuation level set to?

**Comment by britechguy** [comment]
Please tell me exactly what you want me to look at, and where, as far as punctuation level goes, and I'll be happy to.

I will say here what I said on the NVDA group, though:  That should never make a damned bit of difference in a settings dialog.  Settings dialogs should be their own little worlds not subject to things like punctuation level changes, dictionary processing, etc.  You have to have an isolated environment that you are certain is 100% the same for each and every user regardless of what they may have tweaked in order to make their day-to-day use of NVDA more to their liking.  To do otherwise ensures that issues like this can and will occur, and they should never be able to occur (in that way) by intentional design.

**Comment by lukaszgo1** [comment]
The reason why I was unable to reproduce this was indeed due to my symbol level - it was set to "all"  and in that case this issue does not occur.
However you said:

>     4. NVDA reads, "Symbols list in-word semi replacement colon tick semi level colon all preserve colon only below symbol's level 1 of 9"
>        Note: If I down arrow to the standalone single quote, rather than the in-word single quote, NVDA reads:  "semi Replacement colon tick semi Level colon all semi Preserve colon never  2 of 9"


In this example the symbol for the standard single quote is also ignored, so these two examples are identical IMO.

**Comment by britechguy** [comment]
I agree that the skipping of the actual tick/bang character announcement is an identical issue, as is the skipping of just tick whether in the in-word example or standalone.  And what's even more fun is that in speech viewer the character is absolutely there, it's just not announced by NVDA when working in the dialog.  It's very, very strange.

I only caught this because I did a sort of double-take after what I could see and what I could hear did not match each other, and that's a great rarity.  But I almost missed it because all of those semis and colons are being announced without being visible anywhere.  Being a coder (in the past) I recognized that these are separator characters, but I still have absolutely no idea why they would ever be announced under any circumstance in this specific context.  They're extraneous auditory clutter that makes it much harder to focus on what it is you're actually trying to tweak.

**Comment by seanbudd** [comment]
This may be blocked by #10855

**Comment by seanbudd** [comment]
Potentially related to #13924

**Comment by britechguy** [comment]
I'd bet my bottom dollar that this and #13924 share the same roots.  And the issue with punctuation level playing in is a problem in any instance when it's essential that all characters be announced.

There is a need to establish, and clearly, where punctuation level should be applied, and where it should not be applied.  When it comes to dialog boxes or announcing controls, it shouldn't.

**Comment by britechguy** [comment]
This remains an issue in April 2025.  I have the issue, with a bit less verbosity than when originally reported, when using the Symbol pronunciation dialog for the characters:
!
;
?
but by contrast, not for:
#

But when moving through the symbol list, Symbol column, for at least the 3 noted symbols (there are likely to be others) the actual name of the symbol, as currently replaced, is never announced.  This is hugely problematic.

**Comment by britechguy** [comment]
I've really begun to suspect that NVDA is using it's own symbol punction level settings, at whatever their current levels are, and that's sometimes preventing the actual pronunciation of the symbol you're dealing with when working in the Symbol Pronunciation (English) dialog.   If that is happening it's a bug.  There should never be a reason, when in the actual symbol pronunciation dialog, in the symbols list, that the actual symbol character in column 1 is not announced as itself, regardless of whether or not it might be announced in other contexts.  That way lies madness.

If you're in the Symbol pronunciation dialog, the symbol itself, in the first column, or anywhere else it appears in an editable field, must be announced.  Otherwise, it's impossible to know exactly what it is you're dealing with.

**Comment by britechguy** [comment]
@Adriani90 

Thanks for the confirmation this is present in all language localizations.  I reported exactly what I see when I open the Symbol pronunciation dialog, which includes "(English)," but I suspected this applied to all language localizations, but could not state that unequivocally.

Every time I deal with this dialog and a symbol/punctuation itself is not announced when I'm in the Symbol column (and can see it, in my case) my immediate reaction is WTF?!!  I cannot imagine how confusing this is to anyone trying to change the pronunciation of a symbol when that symbol is not read in either the filter box nor in any symbol column in which it's present due to its existing symbol level.

The filter edit box and the symbol edit box both need to be handled as exception conditions where regardless of what NVDA would do based on symbol level anywhere else, it's overridden such that the symbol in question is always announced in those two places.  It's the only logical thing to do, as an end user must know what it is they're working with no matter what its level is set to elsewhere.
